### DISCUSSION TITLE
Installing on linux

### DISCUSSION THREAD

**Discussion by madmodderx** [edward-ly/stepmania-song-manager#10] [Q&A] [answered]
Pretty big Linux noob here, I got it installed on my windows PC and I'm trying to figure out how to install it on my dedi-stepmania Linux box. 
Downloaded the yml, I go to open it and it suggests just text editor.
Google has offered me very little in the way of installing via yml, YouTube for an install tut resulted in nothing more than how to code in yml. Running latest Ubuntu.

I'm at the point where I don't know what I'm supposed to be searching for to install this.

Installed yarn and went to the dir with the yml
Yarn
Got no scenario error message

**Comment by madmodderx** [comment]
Progress, thought i installed yarn, apparently not properly
gwtting new error messages
```
yarn install v1.22.5
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
error @quasar/app@3.0.1: The engine "node" is incompatible with this module. Expected version ">= 12.22.1". Got "10.19.0"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by edward-ly** [comment, marked as answer]
If you just want to use the app, just head [here](https://github.com/edward-ly/stepmania-song-manager/releases/latest) and download one of the `.AppImage` files (*not* `.yml` files), then open it as an executable. No installation required.

**Reply by madmodderx** [reply]
So, it's still not running... Moved it to the desktop, ran in console
`./StepMania-Song-Manager-0.3.1-i386.AppImage --appimage-extract-and-run
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/.DirIcon
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/AppRun
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/LICENSE.electron.txt
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/LICENSES.chromium.html
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/chrome-sandbox
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/chrome_100_percent.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/chrome_200_percent.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/icudtl.dat
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/libEGL.so
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/libGLESv2.so
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/libffmpeg.so
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/libvk_swiftshader.so
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/libvulkan.so
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/am.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/ar.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/bg.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/bn.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/ca.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/cs.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/da.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/de.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/el.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/en-GB.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/en-US.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/es-419.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/es.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/et.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/fa.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/fi.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/fil.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/fr.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/gu.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/he.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/hi.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/hr.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/hu.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/id.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/it.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/ja.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/kn.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/ko.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/lt.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/lv.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/ml.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/mr.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/ms.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/nb.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/nl.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/pl.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/pt-BR.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/pt-PT.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/ro.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/ru.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/sk.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/sl.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/sr.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/sv.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/sw.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/ta.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/te.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/th.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/tr.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/uk.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/vi.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/zh-CN.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/locales/zh-TW.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/resources
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/resources/app-update.yml
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/resources/app.asar
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/resources.pak
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/snapshot_blob.bin
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/stepmania-song-manager
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/stepmania-song-manager.desktop
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/stepmania-song-manager.png
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/swiftshader
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/swiftshader/libEGL.so
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/swiftshader/libGLESv2.so
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/lib
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/lib/libXss.so.1
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/lib/libXtst.so.6
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/lib/libappindicator.so.1
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/lib/libgconf-2.so.4
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/lib/libindicator.so.7
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/lib/libnotify.so.4
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/share
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/share/icons
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/share/icons/hicolor
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/share/icons/hicolor/512x512
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/share/icons/hicolor/512x512/apps
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/usr/share/icons/hicolor/512x512/apps/stepmania-song-manager.png
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/v8_context_snapshot.bin
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/vk_swiftshader_icd.json
/tmp/appimage_extracted_1509f1f62afc3dfd2281bc82a8111ef5/stepmania-song-manager: error while loading shared libraries: libgobject-2.0.so.0: cannot open shared object file: No such file or directory
`
Yes, i did change the properties and permissions with a 'chmod'

**Reply by edward-ly** [reply]
Ah, I just remembered that execute permissions are not on by default, but that's all that is required to run the app. Either `chmod a+x` or `chmod 775` should be enough.

And after that, I don't think `--appimage-extract-and-run` is necessary at all. Just `./StepMania-Song-Manager-0.3.1-i386.AppImage` should be fine.
